src/service/chat.js

```diff
@@ -161,11 +161,55 @@ const getOrCreateAnonymousChannel = async (userId) => {
   }
 };
 
+const moveChatToSigned = async ({oldChannelId, targetUserId, source}) => {
+  try {
+    const response = await api.post('/chat/move-to-sign', {oldChannelId, targetUserId, source});
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `moveChatToSigned` function could be improved. Currently, if the response status is not 200, it rejects with the entire response data which might contain sensitive information. It would be better to reject with a specific error message. Also, the catch block should log the error for debugging purposes.

```diff
-    if (response.status === 200) {
-      return Promise.resolve(response.data);
-    }
-    return Promise.reject(response.data);
+    if (response.status !== 200) {
+      throw new Error('Failed to move chat to signed');
+    }
+    return response.data;
```

```diff
-    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-    return Promise.reject(e);
+    console.error(e);
+    if (e?.response?.data?.message) {
+      throw new Error(e.response.data.message);
+    }
+    throw e;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->